engine.php

```diff
@@ -197,7 +197,16 @@ private function parse_master_response($data, &$index, &$count, &$servers)
             {
                 $ip = [];
                 foreach ( range(0, 3) as $i )
+                {
                     $ip[] = (string)ord($nextbyte());
+                    // ipv4 address field starts with '\'
+                    // but serverResponse packet also ends with '\'
+                    // if end of packet is reached, do not parse ipv4
+                    if ( feof($buffer) )
+                        break;
+                }
+                if ( feof($buffer) )
```

Comment:
This check is not needed since it occurs immediately after the last one.

Reply:
The first check is needed to break the `foreach` loop to not read four `nextbytes` of an ipv4 when first `nextbyte` is EOF, the second check is needed to break the `for` loop to not read two more `nextbytes` of a port when first nextbyte is EOF. And in any way the second check is needed to carry the `break` from the inner loop to the outer loop.

Removing this second check makes the bug reappear because the ip is still parsed by later code.

Reply:
Right... maybe return instead of break so it doesn't have to be repeated twice?